docs/API.md

```diff
@@ -0,0 +1,267 @@
+
+### analytics (`APP_ANALYTICS_CONFIGURED`)
+
+The `analytics` phase initializes Segment and configures @edx/frontend-analytics.
+
+### beforeReady (`APP_BEFORE_READY`)
+
+The `beforeReady` phase has no default behavior.
+
+### ready (`APP_READY`)
+
+The `ready` phase has no default behavior.  This is the phase where an application's interface would generally be shown to the user.
+
+### error (`APP_ERROR`)
+
+The `error` phase logs (to loggingService) whatever error occurred to put the app in an error state.  This is the phase where an application would generally show an error message for an unexpected error to the user.
+
+Note that the error which caused the application to transition to the `error` phase is available at `App.error`.  It is also passed as data to any subscribers to the `APP_ERROR` event.
+
+## `App`
+
+The `App` class is a singleton with static methods.  This is so that it can be used throughout a consuming application merely by importing it:
+
+```
+import { App } from '@edx/frontend-base';
+```
+
+`App` provides a few different methods and properties:
+
+### `App.history`
+
+The application's browser history object.  See the [history documentation](https://github.com/ReactTraining/history/blob/master/docs/GettingStarted.md) for more information.
+
+### `App.authenticatedUser`
+
+A reference to an object containing information about the currently authenticated user.
+
+```
+{
+  userId: <AUTHENTICATED USER's USER ID>,
+  username: <AUTHENTICATED USER's USERNAME>,
+  roles: [<ROLE ONE>, <ROLE TWO>],
+  administrator: <true|false>,
+}
+```
+### `App.decodedAccessToken`
+
+A reference to an object containing the raw, decoded JWT access token.  This is the snake_case source for the data in `App.authenticatedUser`.
+
+### `App.error`
+
+An error object caught by `App.initialize` if an error occurred during initialization.  Useful for error handlers.  Note: This is not populated for errors outside of application initialization.  It's merely a convenience to allow `APP_ERROR` subscribers and `error` overrideHandlers to access the error that put the application in the error phase.
+
+### `App.initialize({ messages, loggingService, overrideHandlers, custom })`
+
+The `App.initialize` method takes an options object with four possible keys:
+
+#### messages
+
+A frontend-i18n-compatible messages object, or an array of such objects.  If an array is provided, duplicate keys are resolved with the last-one-in winning.
+
+#### loggingService
+
+The logging service to be used by @edx/frontend-logging.  It defaults to `NewRelicLoggingService`.
+
+#### overrideHandlers
+
+An optional object of `overrideHandlers` which can be used to replace the default behavior of any part of the startup sequence.  It can also be used to add additional initialization behavior before or after the rest of the sequence.
+
+An example in which we override the authentication handling:
+
+```
+App.initialize({
+  overrideHandlers: {
+    authentication: () => {
```

Comment:
I thought maybe the defaultHandler could be passed to the overridden handler, but probably that is something that could be left for if anyone actually needs that.

Reply:
The default handlers are exported/available to be imported and used in the overrides, fwiw.  That said, this doc is wrong, in that a reference to the `App` is passed to the override handlers!